arith/gf2/src/gf2x64.rs

```diff
@@ -0,0 +1,301 @@
+use std::ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub, SubAssign};
+
+use arith::{Field, FieldSerde, FieldSerdeResult, SimdField};
+
+use super::GF2;
+
+#[derive(Debug, Clone, Copy, Default, PartialEq)]
+pub struct GF2x64 {
+    pub v: u64,
+}
+
+impl FieldSerde for GF2x64 {
+    const SERIALIZED_SIZE: usize = 8;
+
+    #[inline(always)]
+    fn serialize_into<W: std::io::Write>(&self, mut writer: W) -> FieldSerdeResult<()> {
+        writer.write_all(self.v.to_le_bytes().as_ref())?;
+        Ok(())
+    }
+
+    #[inline(always)]
+    fn deserialize_from<R: std::io::Read>(mut reader: R) -> FieldSerdeResult<Self> {
+        let mut u = [0u8; Self::SERIALIZED_SIZE];
+        reader.read_exact(&mut u)?;
+        Ok(GF2x64 {
+            v: u64::from_le_bytes(u),
+        })
+    }
+
+    #[inline]
+    fn try_deserialize_from_ecc_format<R: std::io::Read>(_reader: R) -> FieldSerdeResult<Self> {
+        unimplemented!("We don't have serialization in ecc for gf2x64")
+    }
+}
+
+impl Field for GF2x64 {
+    const NAME: &'static str = "Galois Field 2 SIMD 64";
+
+    const SIZE: usize = 8;
+
+    const FIELD_SIZE: usize = 1;
+
+    const ZERO: Self = GF2x64 { v: 0 };
+
+    const ONE: Self = GF2x64 { v: !0u64 };
+
+    const INV_2: Self = GF2x64 { v: 0 }; // NOTE: should not be used
+
+    #[inline(always)]
+    fn zero() -> Self {
+        GF2x64::ZERO
+    }
+
+    #[inline(always)]
+    fn one() -> Self {
+        GF2x64::ONE
+    }
+
+    #[inline(always)]
+    fn is_zero(&self) -> bool {
+        self.v == 0
+    }
+
+    #[inline(always)]
+    fn random_unsafe(mut rng: impl rand::RngCore) -> Self {
+        GF2x64 { v: rng.next_u64() }
+    }
+
+    #[inline(always)]
+    fn random_bool(mut rng: impl rand::RngCore) -> Self {
+        GF2x64 { v: rng.next_u64() }
+    }
+
+    #[inline(always)]
+    fn exp(&self, exponent: u128) -> Self {
```

Comment:
I think `exp` should simply return `*self`, regardless of what `exponent` we have. The same problem exists for previous implementation of `exp` in `GF2x8`.

Reply:
why? the exp should align with gf2 impl, as it reflects simd - I don't get why each cell of gf2 maps to self regardless of exponent. 

elaborate more.

Reply:
so long as e is non zero.

Reply:
oh, you're right...

Reply:
ok - we are both right partly! the implementation is still wrong in a way - that the condition should branch into 2 conditioned that exponent is 0 or not - I will work on it, now I am focusing on neon thing...

Reply:
The condition is a little confusing. It seems $0^2$ would equal $1$ without any restriction of the exponent.

Reply:
Or probably it's ok to simply return `*self` since $0^0$ is not well-defined anyway. I don't know.

Reply:
see https://github.com/PolyhedraZK/Expander/pull/142/commits/33e9d8cc26ea048eb1be0e88ebd67d1345615ad4 - I am trying to cover all possible cases here...